program-test/src/lib.rs

```diff
@@ -156,7 +153,6 @@ pub fn invoke_builtin_function(
 
     // Re-fetch the instruction context. The previous reference may have been
     // invalidated due to the `set_invoke_context` in a CPI.
-    let transaction_context = &invoke_context.transaction_context;
```

Comment:
We can't remove this line. `set_invoke_context` overwrites `invoke_context` with unsafe code, so there's a possibility of invalidating the previous reference to `transaction_context`. The rest is good though!

Reply:
https://github.com/solana-labs/solana/pull/26507 added this logic, but unfortunately figuring out the issue is lost in a slack chat